### ISSUE TITLE
CI is failing with out of space

### ISSUE THREAD

**Issue by jsturtevant** [containerd/runwasi#252]
It manifests in several different ways but looks something like:

```
 Warning  FailedCreatePodSandBox  43s   kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd container: failed to create prepare snapshot dir: mkdir /var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/new-3941906082/fs: no space left on device
```

**Comment by jsturtevant** [comment]
Happening on 20.04: https://github.com/containerd/runwasi/actions/runs/5905949802/job/16021228149?pr=238

**Comment by jsturtevant** [comment]
another example: https://github.com/containerd/runwasi/actions/runs/5897401337/job/15997018758?pr=248

**Comment by jsturtevant** [comment]
this is failing on main too https://github.com/containerd/runwasi/actions/runs/5919858042

**Comment by Mossaka** [comment]
I am using PR #261 to inspect disk usage for the CI. This is the latest failed run on ubuntu 20.04 for wasmtime integration tests. What's interesting to me is that the /dev/root is almost full after the last step.

![image](https://github.com/containerd/runwasi/assets/5447827/6c1fe6aa-f883-487d-a0d5-52558510f1dd)

**Comment by jprendes** [comment]
I guess that's expected as that's the root of the filesystem. Can you try adding `df` again? Maybe with `--max-depth=1`?

**Comment by Mossaka** [comment]
There is no `max-depth` for `df`. You meant `du`?

**Comment by jprendes** [comment]
Yes, sorry,  `du`

**Comment by Mossaka** [comment]
The result of running `du`

![image](https://github.com/containerd/runwasi/assets/5447827/6d5b38c9-898b-4462-8875-bda832537f45)

**Comment by jprendes** [comment]
We could use https://github.com/insightsengineering/disk-space-reclaimer to reclaim up to (they claim) 34 Gb of storage in the runners.

**Comment by jsturtevant** [comment]
Interesting! Is it possible our Rust build cache needs to be cleaned up too?

**Comment by Mossaka** [comment]
Docker build cache:
```
Build Cache     20        0         12.55GB   12.55GB
```

**Comment by Mossaka** [comment]
I just merged in #263 . Hopefully it will unblock all other PRs.

**Comment by jprendes** [comment]
Did you mean to keep this issue open?

**Comment by jsturtevant** [comment]
thanks @Mossaka

**Comment by Mossaka** [comment]
It looks like the CI still failed with a different error:

```
  Warning  FailedScheduling  81s   default-scheduler  0/1 nodes are available: 1 node(s) had taint {node.kubernetes.io/not-ready: }, that the pod didn't tolerate.
```

**Comment by Mossaka** [comment]
See this ran: https://github.com/containerd/runwasi/actions/runs/5944133273/job/16120966602?pr=238

**Comment by jprendes** [comment]
Is there any reason why wasmtime is tested with `kind` while wasmedge is tested with `k3s`?
I'm not very familiat with k8s, but it seems like `kind` is more heavy weight than k3s.

**Comment by jsturtevant** [comment]
out of disk was consistently happening on 20.04, this new error is on 22.04,  lets create a new issue to track that?

**Comment by jsturtevant** [comment]
> Is there any reason why wasmtime is tested with `kind` while wasmedge is tested with `k3s`?
> I'm not very familiat with k8s, but it seems like `kind` is more heavy weight than k3s.

not sure, kind is widely used for CI
